File: Prak2-Inheritance/Expression/NegativeExpression.cpp
/* Nama         : Gregorius Jovan Kresnadi */
/* NIM          : 13518135 */
/* Tanggal      : 13-02-2020 */

#include "NegativeExpression.hpp"
#include <iostream>

using namespace std;

NegativeExpression::NegativeExpression(Expression* x) : UnaryExpression(x) {
    //
}

int NegativeExpression::solve() {
	return x->solve() * -1;

}

File: PrakJ1-Intro/T/2/NegativeExpression.java
// Nama         : Gregorius Jovan Kresnadi
// NIM          : 13518135
// Tanggal      : 12-03-2020
// File 		: NegativeExpression.java

class NegativeExpression implements Expression {
	protected Expression x;

	public NegativeExpression(Expression x) {
		this.x = x;
	}

	public int solve() {
		return x.solve() * (-1);
	}
}
